rfd/0189-windows-access-guide.md

```diff
@@ -0,0 +1,105 @@
+---
+authors: Dave Sudia (david.sudia@goteleport.com)
+state: draft
+---
+
+# RFD 0189 - Windows Desktop Access guide
+
+## Required Approvers
+
+* Engineering: @r0mant
+* Product: @xinding33
+
+## Overview
+
+### Problem
+
+* Currently only 15% of users convert from going to the docs to starting a session
+* We need better insight into drop-off points: we only know they clicked out to the docs then started a session and don’t know where they’re struggling
+
+### Other data/context
+
+Historically we have more local user sessions than AD sessions (last 180 days, 36.3k local vs 30.1k AD). This may be skewed by us not getting metrics from major self-hosted customers who are likely using AD.
+
+### Appetite/Resources
+
+We want the scope of this project to be doable for a full-stack engineer with some design input, within 6 weeks.
```

Comment:
I don't think this is doable in 6 weeks.

We had a tighter-scoped goal of doing this only for EC2 instances on AWS a couple quarters ago and couldn't manage to fit that in with a dev who was experienced in the Windows domain.

This is more complicated than any of the existing discover flows that I'm aware of, because you need to install and configure Teleport software on at least two servers instead of 1 (the Linux instance running the agent and the target Windows instance).

Remotely installing software on a Windows machine is not something we've ever done, and this is further complicated by the facts that:
- the Windows host has to be rebooted after the software is installed
- the Teleport software that runs on Windows doesn't make any network connections, so it can't phone home and tell us when it's ready

None of this is impossible to solve, but it's tricky and likely requires a lot more than 6 weeks of a fullstack developer's time.

Reply:
@zmb3 I think I needed to be clearer about intention. I don't think we need to make this remotely installing the software. For other guides like SSH we just give them a script and have them run it. I think we should offer some kind of confirmation that the Desktop Service is running, but that can be a manual run of the script, and similarly running the script on the Windows box and restarting it can be manual actions on the user's part, we're just auto-generating the script.

Let me know what you think.